src/main/java/org/spongepowered/api/variant/VariantQueryableCatalogType.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.variant;
+
+import org.spongepowered.api.CatalogType;
+
+import java.util.Optional;
+
+public interface VariantQueryableCatalogType extends CatalogType {
+
+    /**
+     * Gets whether this {@link CatalogType} matches
+     * the specified queries.
+     *
+     * @param queries The queries
+     * @return Matches
+     */
+    boolean matches(VariantQuery<?>... queries);
+
+    /**
+     * Gets the query argument that can be used to match this
+     * {@link CatalogType} with the specified {@link VariantQueryType}.
+     * {@link Optional#empty()} will be returned if the query type
+     * isn't supported by this catalog type.
+     *
+     * @param queryType The query type
+     * @param <T> The argument type
+     * @return The argument type, if the query type is supported
+     */
+    <T> Optional<T> getArgument(VariantQueryType<T> queryType);
```

Comment:
Why is it a query type instead of just `VariantQuery`? We don't call `Key`s `KeyType`.

Reply:
It was to return the argument you would pass in to create the `VariantQuery` from the `VariantQueryType`.